### DISCUSSION TITLE
quasar build error

### DISCUSSION THREAD

**Discussion by mirronovko** [quasarframework/quasar#11161] [CLI - PWA mode] [answered]
quasar.conf.js
```
module.exports = function ( ctx ) {
    return {
        // https://quasar.dev/quasar-cli/supporting-ts
        supportTS: true,
```

run: quasar build
 ```
Build mode........ spa
 Pkg quasar........ v1.15.1
 Pkg @quasar/app... v2.1.14
 Debugging......... no
 Publishing........ no

Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
 Configured browser support (at least 87.59% of global marketshare):
 · Chrome for Android >= 88
 · Firefox for Android >= 83
 · Android >= 81
 · Chrome >= 78
 · Edge >= 85
 · Firefox >= 76
 · iOS >= 10.3
 · Opera >= 68
 · Safari >= 11

 App · Running "@quasar/dotenv" Quasar App Extension...
 App · Reading quasar.conf.js
 App · Extension(@quasar/dotenv): Extending quasar.conf...
App Extension (dotenv): '.env' file missing; skipping
 App · Transpiling JS (Babel active)
 App · Extending SPA Webpack config
 App · Cleaned build artifact: "/app/dist/spa"
 App · Generating Webpack entry point
 App · Compiling with Webpack...
Starting type checking service...
 App · Compiling SPA...
 App · Compiled SPA done in 11426 ms
 App · Added build artifact "/app/dist/spa"

 App · ⚠️  6 errors encountered:

/app/node_modules/vue-router/types/vue.d.ts
ERROR in /app/node_modules/vue-router/types/vue.d.ts(10,5):
TS2717: Subsequent property declarations must have the same type.  Property '$router' must be of type 'VueRouter', but here has type 'VueRouter'.
/app/node_modules/vue-router/types/vue.d.ts
ERROR in /app/node_modules/vue-router/types/vue.d.ts(11,5):
TS2717: Subsequent property declarations must have the same type.  Property '$route' must be of type 'Route', but here has type 'Route'.
/app/node_modules/vue-router/types/vue.d.ts
ERROR in /app/node_modules/vue-router/types/vue.d.ts(17,5):
TS2717: Subsequent property declarations must have the same type.  Property 'router' must be of type 'VueRouter | undefined', but here has type 'VueRouter | undefined'.
/app/node_modules/vue-router/types/vue.d.ts
ERROR in /app/node_modules/vue-router/types/vue.d.ts(18,5):
TS2717: Subsequent property declarations must have the same type.  Property 'beforeRouteEnter' must be of type 'NavigationGuard<V> | undefined', but here has type 'NavigationGuard<V> | undefined'.
/app/node_modules/vue-router/types/vue.d.ts
ERROR in /app/node_modules/vue-router/types/vue.d.ts(19,5):
TS2717: Subsequent property declarations must have the same type.  Property 'beforeRouteLeave' must be of type 'NavigationGuard<V> | undefined', but here has type 'NavigationGuard<V> | undefined'.
/app/node_modules/vue-router/types/vue.d.ts
ERROR in /app/node_modules/vue-router/types/vue.d.ts(20,5):
TS2717: Subsequent property declarations must have the same type.  Property 'beforeRouteUpdate' must be of type 'NavigationGuard<V> | undefined', but here has type 'NavigationGuard<V> | undefined'.

 App · ⚠️  [FAIL] Build failed with 6 errors. Check log above.
```

node v16.13.0
yarn 1.22.15

help to solve the problem

**Comment by hawkeye64** [comment]
@IlCallo @yusufkandemir

**Reply by mirronovko** [reply]
can anyone help me?

**Comment by yusufkandemir** [comment, marked as answer]
Might be related: https://github.com/quasarframework/quasar/issues/11173
TL;DR: make sure you don't have `vue-router` on your dependencies, remove node_modules and the lock file, reinstall your dependencies. If it didn't solve, please share the outputs of `quasar info` and `yarn why router`.

**Reply by mirronovko** [reply]
i removed node_modules, yarn.lock and vue-router. reinstalled the packages. earned. thank you for your answer
